drivers/crypto/crypto_si32.c

```diff
@@ -0,0 +1,1170 @@
+
+	switch (key_len) {
+	case 32:
+		SI32_AES_0->HWKEY7.U32 = key_as_word[7];
+		SI32_AES_0->HWKEY6.U32 = key_as_word[6];
+		__fallthrough;
+	case 24:
+		SI32_AES_0->HWKEY5.U32 = key_as_word[5];
+		SI32_AES_0->HWKEY4.U32 = key_as_word[4];
+		__fallthrough;
+	case 16:
+		SI32_AES_0->HWKEY3.U32 = key_as_word[3];
+		SI32_AES_0->HWKEY2.U32 = key_as_word[2];
+		SI32_AES_0->HWKEY1.U32 = key_as_word[1];
+		SI32_AES_0->HWKEY0.U32 = key_as_word[0];
+		break;
+	default:
+		LOG_ERR("Invalid key len: %" PRIu16, key_len);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int crypto_si32_aes_calc_decryption_key(const struct cipher_ctx *ctx,
+					       uint8_t *decryption_key)
+{
+	uint32_t *decryption_key_word = (uint32_t *)decryption_key;
+	int ret;
+
+	ret = crypto_si32_aes_set_key(ctx->key.bit_stream, ctx->keylen);
+	if (ret) {
+		return ret;
+	}
+
+	LOG_INF("Generating decryption key");
+	/* TODO: How much of this can be removed? */
+	SI32_AES_A_write_xfrsize(SI32_AES_0, 0);
+	SI32_AES_A_enable_error_interrupt(SI32_AES_0);
+	SI32_AES_A_exit_cipher_block_chaining_mode(SI32_AES_0);
+	SI32_AES_A_exit_counter_mode(SI32_AES_0);
+	SI32_AES_A_exit_bypass_hardware_mode(SI32_AES_0);
+	SI32_AES_A_select_xor_path_none(SI32_AES_0);
+	SI32_AES_A_select_software_mode(SI32_AES_0);
+	SI32_AES_A_select_encryption_mode(SI32_AES_0);
+	SI32_AES_A_enable_key_capture(SI32_AES_0);
+
+	for (unsigned int i = 0; i < 4; i++) {
+		SI32_AES_A_write_datafifo(SI32_AES_0, 0x00000000);
+	}
+
+	SI32_AES_A_clear_operation_complete_interrupt(SI32_AES_0);
+	SI32_AES_A_start_operation(SI32_AES_0);
+	while (!SI32_AES_A_is_operation_complete_interrupt_pending(SI32_AES_0)) {
+		/* This should not take long */
+	}
+
+	for (unsigned int i = 0; i < 4; i++) {
+		SI32_AES_A_read_datafifo(SI32_AES_0);
+	}
+
+	switch (ctx->keylen) {
+	case 32:
+		decryption_key_word[7] = SI32_AES_0->HWKEY7.U32;
+		decryption_key_word[6] = SI32_AES_0->HWKEY6.U32;
+		__fallthrough;
+	case 24:
+		decryption_key_word[5] = SI32_AES_0->HWKEY5.U32;
+		decryption_key_word[4] = SI32_AES_0->HWKEY4.U32;
+		__fallthrough;
+	case 16:
+		decryption_key_word[3] = SI32_AES_0->HWKEY3.U32;
+		decryption_key_word[2] = SI32_AES_0->HWKEY2.U32;
+		decryption_key_word[1] = SI32_AES_0->HWKEY1.U32;
+		decryption_key_word[0] = SI32_AES_0->HWKEY0.U32;
+		break;
+	default:
+		LOG_ERR("Invalid key len: %" PRIu16, ctx->keylen);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int crypto_si32_aes_set_key_size(const struct cipher_ctx *ctx)
+{
+	switch (ctx->keylen) {
+	case 32:
+		SI32_AES_A_select_key_size_256(SI32_AES_0);
+		break;
+	case 24:
+		SI32_AES_A_select_key_size_192(SI32_AES_0);
+		break;
+	case 16:
+		SI32_AES_A_select_key_size_128(SI32_AES_0);
+		break;
+	default:
+		LOG_ERR("Invalid key len: %" PRIu16, ctx->keylen);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static void assert_dma_settings_common(struct SI32_DMADESC_A_Struct *channel_descriptor)
+{
+	if (!IS_ENABLED(CONFIG_ASSERT)) {
+		ARG_UNUSED(channel_descriptor);
+	}
```

Comment:
I think most places that the variable is just used in asserts we just `ARG_UNUSED` unconditioanlly.

Reply:
done